app/Models/CampaignDashboardWidget.php

```diff
@@ -304,7 +304,7 @@ public function randomEntity()
     {
         $entityType = $this->conf('entity');
         $entityTypeID = null;
-        if ($entityType != 'any') {
+        if ($entityType && $entityType != '') {
```

Comment:
Isn't the data saved as null? In that case, !empty() would be cleaner and easier to read